ert/ensemble_evaluator/activerange.py

```diff
@@ -0,0 +1,136 @@
+        if is_active:
+            if start is None:  # begin tracking a range
+                start = i
+            assert start is not None
+        else:
+            if start is not None:  # store the range and stop tracking
+                store_range(start, i)
+                start = None
+            assert start is None
+    if start is not None:  # complete the last range if any
+        store_range(start, len(mask))
+    return ",".join(ranges)
+
+
+def rangestring_to_mask(rangestring: str, length: int) -> List[bool]:
+    """Convert a string specifying ranges of elements, and the number of elements,
+    into a list of booleans. The ranges are end-inclusive."""
+    mask = [False] * length
+    if rangestring == "":
+        # An empty string means all realizations deactivated. Note that an
+        # ActiveRange-typed instance being None means the opposite
+        return mask
+    for _range in rangestring.split(","):
+        if "-" in _range:
+            if len(_range.strip().split("-")) != 2:
+                raise ValueError(f"Wrong range syntax {_range}")
+            start, end = map(int, _range.strip().split("-"))
+            print(end)
+            if end + 1 > length:
+                raise ValueError(
+                    f"Range endpoint {end} is beyond the mask length {length} "
+                )
+            mask[start : end + 1] = [True] * (end + 1 - start)
+        elif _range:
+            if int(_range) + 1 > length:
+                raise ValueError(
+                    f"Realization index {_range} is beyond the mask length {length} "
+                )
+            mask[int(_range)] = True
+    return mask
+
+
+class ActiveRange:
+    def __init__(
+        self,
+        mask: Optional[List[bool]] = None,
+        rangestring: Optional[str] = None,
+        length: Optional[int] = None,
+    ):
+        if mask is None and rangestring is None and length is None:
+            raise ValueError("Supply mask or rangestring and length to ActiveRange.")
+        if mask is None:
+            if length is not None and rangestring is not None:
+                self._mask = rangestring_to_mask(rangestring=rangestring, length=length)
+            else:
+                raise ValueError("Must supply both rangestring and length")
+        else:
+            if rangestring is not None:
+                raise ValueError("Can't supply both mask and rangestring")
+            if length is not None:
+                assert length == len(mask)
+            self._mask = mask
+
+    def get_mask(self) -> List[bool]:
+        return list(self._mask)
+
+    def get_rangestring(self) -> str:
+        return mask_to_rangestring(self._mask)
+
+    def __len__(self) -> int:
+        return len(self._mask)
+
+    def __repr__(self) -> str:
+        return mask_to_rangestring(self._mask) + f" length={len(self._mask)}"
+
+    @classmethod
+    def validate_rangestring(cls, rangestring: str) -> str:
```

Comment:
Just wondering here of why not to use regex? Eg. `re.match("\d+-\d+", rangestring)`

Reply:
Regex could have been used, but the validator currently gives more precise error messages due to not using a regex.